nixos/modules/services/system/hot-resize.nix

```diff
@@ -0,0 +1,84 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.hotResize;
+in
+{
+  options.services.hotResize = {
+    enable = lib.mkEnableOption "Enable hot resize service for filesystems";
+
+    device = lib.mkOption {
+      type = lib.types.str;
+      example = "/dev/sda1";
+      description = "Device to resize";
+    };
+
+    fsType = lib.mkOption {
+      type = lib.types.enum [
+        "ext4"
+        "xfs"
+        "btrfs"
+      ];
+      default = "ext4";
+      description = "Filesystem type (supported: ext4, xfs, btrfs)";
+    };
+
+    mountPoint = lib.mkOption {
+      type = lib.types.str;
+      example = "/";
+      description = "Mount point of the filesystem to resize";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.hotResize = {
+      description = "Hot resize service for filesystems";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = false;
+        ExecStart = pkgs.writeShellScript "resize-fs" ''
+          ${lib.getExe' pkgs.parted "partprobe"}
+
+          DISK=$(echo ${cfg.device} | sed 's/[0-9]*$//')
+          PART_NUM=$(echo ${cfg.device} | grep -o '[0-9]*$')
+
+          echo "Growing partition..."
+          ${lib.getExe' pkgs.cloud-utils "growpart"} $DISK $PART_NUM || true
```

Comment:
I am concerned about the use of `|| true` here. Why do we expect this to fail? Considering how dangerous disk operations are, I'd rather have a more robust system here.